server/graphql/v2/query/collection/OrdersCollectionQuery.ts

```diff
@@ -180,7 +180,15 @@ export const OrdersCollectionResolver = async (args, req: express.Request) => {
   const searchTermConditions = buildSearchConditions(args.searchTerm, {
     idFields: ['id'],
     slugFields: ['$fromCollective.slug$', '$collective.slug$'],
-    textFields: ['$fromCollective.name$', '$collective.name$', 'description'],
+    textFields: [
+      '$fromCollective.name$',
+      '$collective.name$',
+      'description',
+      'data.ponumber',
+      'data.memo',
```

Comment:
:no_entry_sign: The `memo` is for admins only; we can't allow searching it for everyone.